app/src/main/java/org/stepic/droid/core/presenters/InvitationPresenter.kt

```diff
@@ -14,17 +13,12 @@ class InvitationPresenter
 @Inject constructor(
         private val threadPoolExecutor: ThreadPoolExecutor,
         private val mainHandler: MainHandler,
-        private val sharedPreferenceHelper: SharedPreferenceHelper,
         private val analytic: Analytic) : PresenterBase<InvitationView>() {
 
     fun needShowInvitationDialog(course: Course) {
         threadPoolExecutor.execute {
-            if (sharedPreferenceHelper.isInvitationWasDeclined) {
-                analytic.reportEvent(Analytic.Interaction.INVITATION_PREVENTED)
-            } else {
-                analytic.reportEvent(Analytic.Interaction.SHOW_MATERIAL_DIALOG_INVITATION)
-                mainHandler.post { view?.onShowInvitationDialog(course) }
-            }
+            analytic.reportEvent(Analytic.Interaction.SHOW_MATERIAL_DIALOG_INVITATION)
```

Comment:
threadPoolExecutor and mainHandler are not needed anymore. Let's remove it and also remove dependencies